apis/engine/v1alpha1/azure_role_types.go

```diff
@@ -0,0 +1,127 @@
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	appcat "kmodules.xyz/custom-resources/apis/appcatalog/v1alpha1"
+)
+
+const (
+	ResourceKindAzureRole = "AzureRole"
+	ResourceAzureRole     = "azurerole"
+	ResourceAzureRoles    = "azureroles"
+)
+
+// +genclient
+// +k8s:openapi-gen=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AzureRole
+type AzureRole struct {
+	metav1.TypeMeta   `json:",inline,omitempty"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              AzureRoleSpec   `json:"spec,omitempty"`
+	Status            AzureRoleStatus `json:"status,omitempty"`
+}
+
+type AzureSecretType string
+
+const (
+	AzureClientSecret = "clientSecret"
+)
+
+// AzureRoleSpec contains connection information, Azure role info, etc
+type AzureRoleSpec struct {
+	AuthManagerRef *appcat.AppReference `json:"authManagerRef,omitempty"`
+	Config         *AzureConfig         `json:"config"`
+
+	// ref:
+	// - https://www.vaultproject.io/api/secret/azure/index.html#parameters
+
+	// List of Azure roles to be assigned to the generated service principal.
+	// The array must be in JSON format, properly escaped as a string
+	AzureRoles string `json:"azureRoles,omitempty"`
+
+	// Application Object ID for an existing service principal
+	// that will be used instead of creating dynamic service principals.
+	// If present, azure_roles will be ignored.
+	ApplicationObjectID string `json:"applicationObjectID, omitempty"`
+
+	// Specifies the default TTL for service principals generated using this role.
+	// Accepts time suffixed strings ("1h") or an integer number of seconds.
+	// Defaults to the system/engine default TTL time.
+	TTL string `json:"ttl, omitempty"`
+
+	// Specifies the maximum TTL for service principals
+	// generated using this role. Accepts time suffixed strings ("1h")
+	// or an integer number of seconds. Defaults to the system/engine max TTL time.
+	MaxTTL string `json:"maxTTL, omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type AzureRoleList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata, omitempty"`
+
+	// Items is a list of AzureRole objects
+	Items []AzureRole `json:"items, omitempty"`
+}
+
+// ref:
+//	- https://www.vaultproject.io/api/secret/azure/index.html#configure-access
+
+// AzureConfig contains information to communicate with Azure
+type AzureConfig struct {
+	// The subscription id for the Azure Active Directory.
+	// +required
+	SubscriptionID string `json:"subscriptionID"`
+
+	// The tenant id for the Azure Active Directory.
+	// +required
+	TenantID string `json:"tenantID"`
+
+	// The OAuth2 client id to connect to Azure.
+	// +optional
+	ClientID string `json:"clientID, omitempty"`
+
+	// The OAuth2 client secret to connect to Azure.
+	// This value will be provided by kubernetes secret
+	//	secret.Data:
+	//		clientSecret: <value>
+	ClientSecret string `json:"clientSecret, omitempty"`
```

Comment:
This is confidential information. We need to use a `Secret` to pass this.

Reply:
Solved. Credential information are moved to secret